client/src/components/VerifyInfo/index.tsx

```diff
@@ -0,0 +1,102 @@
+import type { WithStyles, Theme } from '@material-ui/core/styles';
+
+import React, { useState, useContext } from 'react';
+import { withStyles, createStyles } from '@material-ui/core/styles';
+import { Fade, Box, Typography } from '@material-ui/core';
+import { UserContext } from 'context/user/state';
+import UserInfoForm from 'components/forms/UserInfo';
+import BWButton from 'components/buttons/BWButton';
+import { isName, isStudentNumber, isUWEmail } from 'components/forms/utils';
+
+const styles = (theme: Theme) =>
+  createStyles({
+    subtitle: {
+      marginTop: theme.spacing(1),
+    },
+    actions: {
+      marginTop: theme.spacing(4),
+      padding: `0 ${theme.spacing(4)}px`,
+    },
+  });
+
+function VerifyInfo({ classes }: WithStyles<typeof styles>) {
+  const { user, setUser } = useContext(UserContext);
+  const [firstName, setFirstName] = useState(user.firstName);
+  const [lastName, setLastName] = useState(user.lastName);
+  const [studentNumber, setStudentNumber] = useState(
+    user.studentNumber.toString(),
+  );
+  const [email, setEmail] = useState(user.email);
+  const [semester, setSemester] = useState(user.semester);
+  const [faculty, setFaculty] = useState(user.faculty);
+
+  const [triedToSubmit, setTriedToSubmit] = useState(false);
+
+  const onSubmit = () => {
+    setTriedToSubmit(true);
+    if (
+      isName(firstName) &&
+      isName(lastName) &&
+      isStudentNumber(studentNumber) &&
+      isUWEmail(email) &&
+      !!semester &&
+      !!faculty
+    ) {
+      setUser({
+        ...user,
+        firstName,
+        lastName,
+        studentNumber: parseInt(studentNumber),
+        email,
+        semester,
+        faculty,
+      });
+    }
+  };
+
+  return (
+    <div>
+      <Fade appear in timeout={800}>
+        <Box display="flex" flexDirection="column" alignItems="center">
+          <Typography variant="h3" align="center">
+            Verify Your Information
+          </Typography>
+          <div className={classes.subtitle}>
```

Comment:
just interested

I really like boxes here so that you can see the layout of the JSX without having to scroll up into the classes.
There's also the argument that it makes things more cluttered and splits styles into two places.

what do u think?

Reply:
Yea it's just been a habit of mine to use divs. Boxes look cool. 
Will try to use them from now on.
But we can also add a refactor to swap all possible divs for boxes